lib/pigeon.ex

```diff
@@ -119,19 +119,30 @@ defmodule Pigeon do
     ref = :erlang.make_ref()
     on_response = fn x -> send(myself, {:"$push", ref, x}) end
     notification = put_on_response(notification, on_response)
-
-    worker_info =
-      with {:ok, worker_pid} when is_pid(worker_pid) <- push_async(pid, notification) do
-        GenServer.call(worker_pid, :info)
-      end
+    t0 = :erlang.monotonic_time()
+    start_response = push_async(pid, notification)
+    worker_pid = get_worker_pid(start_response)
+    worker_info = GenServer.call(worker_pid, :info)
+    timeout = calculate_timeout(timeout, worker_info)
 
     receive do
-      {:"$push", ^ref, x} -> x |> put_worker_info(worker_info)
+      {:"$push", ^ref, x} ->
+        rtt = :erlang.monotonic_time() - t0
+        GenServer.call(worker_pid, {:update_timing_data, rtt})
```

Comment:
Small suggestion: might be nice to do a cast here instead so that it doesn't block the response back from this call.